src/middlewared/middlewared/service/core_service.py

```diff
@@ -543,18 +543,29 @@ def ping(self):
         """
         return 'pong'
 
-    def _ping_host(self, host, timeout):
-        return run(['ping', '-4', '-w', f'{timeout}', host]).returncode == 0
-
-    def _ping6_host(self, host, timeout):
-        return run(['ping6', '-w', f'{timeout}', host]).returncode == 0
+    def _ping_host(self, version, host, timeout, count=None, interface=None, interval=None):
+        if version == 4:
+            command = ['ping', '-4', '-w', f'{timeout}']
```

Comment:
Just something to note:
If you do `ping -4 -w 2 -c 4`, the return code will not be `0` like you expect (regardless if the 1st 2 packets go through successfully.

Reply:
Understood.  FWIW, adding an appropriate interval to the above would allow all 4 pkts to go thru in the allotted time.